### ISSUE TITLE
share cache between templates with/without the profiler

### ISSUE THREAD

**Issue by fruitl00p** [twigphp/Twig#1742]
Currently the cached Twig templates that had the profiler enabled cannot be used when the profiler is _not_ enabled. Is there any plan / idea (or reasoning) behind the fact that the template cache cannot be shared between the two environments?

The main difference between the internal cache is:
```php
protected function doDisplay(array $context, array $blocks = array())
{
    $__internal_some_random_hash = $this->env->getExtension("profiler");
    __internal_some_random_hash->enter($__internal_some_random_hash_prof = new Twig_Profiler_Profile($this->getTemplateName(), "template", "some_template.twig"));
    // ...the actual parsed template
    $__internal_some_random_hash->leave($__internal_some_random_hash_prof);
    }
```
and ofcourse the ```getDebugInfo()``` is different ;)

In our situation we utilize various on demand profiling and by reusing the cache seems logical doesnt it? Whats the idea behind this?

**Comment by fabpot** [comment]
Templates generated in the cache can never be used safely with a different Twig configuration (options, extensions, ...); the profiler is no exception.

**Comment by fruitl00p** [comment]
That helps, i just would have expected the hashed filename of the cachefile to be different depending on the active options, extensions etc... Now we get fatal errors due to keeping the same cache-directory but changing the loaded extensions ;)

Any chance this might get changed ?

**Comment by fabpot** [comment]
@fruitl00p Are you using Twig with Symfony? Looks similar to symfony/symfony#15318

**Comment by fruitl00p** [comment]
Nope, we're using Twig outside of the symfony ecosystem. We 'bypassed' the situation by creating seperate directories for every new extension we dynamically add to the Twig environment... This is pretty ugly but prevent clashes as before. A more clean solution would be having different filenames depending on the loaded stuff ? That way we can have a single ```$instance->setCache('some_temp_twig_dir');``` and have the ```Twig_Environment``` create different files per configuration...

**Comment by fabpot** [comment]
should be fixed by #1824

**Comment by fruitl00p** [comment]
LGTM, such a simple fix i could've have PR'-ed.. will pay more attention next time ;) thanks!
